os_is_not_exist_matcher.go

```diff
@@ -0,0 +1,32 @@
+package gomegamatchers
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/onsi/gomega/types"
+)
+
+func BeAnOsIsNotExistError() types.GomegaMatcher {
```

Comment:
We took this pattern from the [Gomega documentation](http://onsi.github.io/gomega/#adding-your-own-matchers), and this appears to be the same pattern as [every other](https://github.com/onsi/gomega/blob/3e93f011213c200aadafcb19ed56eb718381cfbd/matchers.go) Gomega matcher.